### DISCUSSION TITLE
Why kratos uses a two-steps logout flow for browser client?

### DISCUSSION THREAD

**Discussion by laplaceliu** [ory/kratos#2526] [Q&A] [answered]
I am very curious that why kratos has designed to use a two-steps logout flow for browser client. I think there must be some reasonable design decisions, but the docs have not mentioned it.

**Comment by aeneasr** [comment, marked as answer]
A two-legged flow is required to protect against [Logout CSRF](https://owasp.org/www-community/attacks/csrf)

**Reply by weizenspreu** [reply]
I've stumbled across this as well. I understand that the CSRF prevention is currently necessary because the actual logout call uses the GET request which can even be triggered with `SameSite=strict` being set for the session cookie. However, wouldn't the necessity of an additional CSRF token go away when the logout instead required a POST request (which I'd expect anyways as the logout leads to a state change)?
